File: C#/219. Contains Duplicate II.cs
public class Solution {
    public bool ContainsNearbyDuplicate(int[] nums, int k) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for(int i=0; i < nums.Length; i++){
            if(dict.ContainsKey(nums[i])){
                int j = dict[nums[i]];
                if(i-j<=k){
                    return true;
                }
            }
            dict[nums[i]]=i;
        }
        return false;
    }
}

File: Java/219. Contains Duplicate II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> m = new HashMap<Integer, Integer>();
        for(int i=0; i < nums.length; i++){
            if(m.containsKey(nums[i])){
                int j = m.get(nums[i]);
                if(i-j<=k){
                    return true;
                }
            }
            m.put(nums[i], i);
        }
        return false;
    }
}
